src/main/java/com/google/api/codegen/java/JavaGapicContext.java

```diff
@@ -120,6 +120,13 @@ public String getTypeName(String typeName) {
   }
 
   /**
+   * Returns the resource name from the entity name.
+   */
+  public String getResourceName(String entityName) {
+    return lowerUnderscoreToUpperCamel(lowerCamelToLowerUnderscore(entityName)) + "Name";
```

Comment:
It seems a bit weird to me but the entiryName can be something like: cryptoKey_path so neither lowerCamelToUpperCamel nor lowerUnderScoreToUpperCamel fit.
What I am doing here (as a work-around) is: cryptoKey_path -> crypto_key_path -> CryptoKeyPath